ql/time/date.cpp

```diff
@@ -842,12 +842,16 @@ namespace QuantLib {
             FormatResetter(std::ostream &out)
                 : out_(&out), flags_(out.flags()), filler_(out.fill()),
                   loc_(out.getloc()) {
+                struct nopunct : std::numpunct<char> {
+                    std::string do_grouping() const {return "";}
+                };
+                std::locale loc (out.getloc(),new nopunct);
```

Comment:
This gives me a compile error on g++ 5.4 (no matching constructor). FWIW, I'm using the `-std=c++03` flag to force C++03 compilation.